src/watertap_contrib/seto/unit_models/surrogate/med_tvc_surrogate.py

```diff
@@ -0,0 +1,1181 @@
+            self.feed_props[0].conc_mass_phase_comp["Liq", "TDS"],
+            to_units=pyunits.mg / pyunits.L,
+        )
+
+        # Set alias for feed water temperature and convert to degree C for surrogate model
+        self.Tin = self.feed_props[0].temperature - 273.15 * pyunits.K
+
+        # Set alias for feed water enthalpy and convert to kJ/kg
+        h_sw = pyunits.convert(
+            self.feed_props[0].enth_mass_phase["Liq"], to_units=pyunits.kJ / pyunits.kg
+        )
+
+        # Set alias for feed water mass flow rate (kg/s)
+        m_f = sum(
+            self.feed_props[0].flow_mass_phase_comp["Liq", j]
+            for j in self.feed_props.component_list
+        )
+
+        """
+        Add block for distillate
+        """
+        self.distillate_props = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of distillate",
+            **tmp_dict
+        )
+
+        # distillate salinity is 0
+        @self.Constraint(doc="distillate salinity")
+        def distillate_s(b):
+            return b.distillate_props[0].conc_mass_phase_comp["Liq", "TDS"] == 0
+
+        # Connect the plant capacity with the distillate mass flow rate
+        self.Capacity = Var(
+            initialize=2000,
+            bounds=(2000, 100000),
+            units=pyunits.m**3 / pyunits.d,
+            doc="Capacity of the plant (m3/day)",
+        )
+
+        @self.Constraint(doc="distillate mass flow rate")
+        def distillate_mfr(b):
+            distillate_mfr = b.distillate_props[0].flow_vol_phase["Liq"]
+            return b.Capacity == pyunits.convert(
+                distillate_mfr, to_units=pyunits.m**3 / pyunits.d
+            )
+
+        # distillate temperature is the same as last effect vapor temperature, which is 10 deg higher than condenser inlet seawater temperature
+        @self.Constraint(doc="distillate temperature")
+        def distillate_temp(b):
+            return (
+                b.distillate_props[0].temperature
+                == self.Tin + (273.15 + 10) * pyunits.K
+            )
+
+        # Set alias for distillate enthalpy and convert to kJ/kg
+        h_d = pyunits.convert(
+            self.distillate_props[0].enth_mass_phase["Liq"],
+            to_units=pyunits.kJ / pyunits.kg,
+        )
+
+        # Set alias for distillate mass (kg/s) flow rate
+        m_d = sum(
+            self.distillate_props[0].flow_mass_phase_comp["Liq", j]
+            for j in self.distillate_props.component_list
+        )
+
+        """
+        Add block for brine
+        """
+        tmp_dict["defined_state"] = False
+
+        self.brine_props = self.config.property_package.state_block_class(
+            self.flowsheet().config.time, doc="Material properties of brine", **tmp_dict
+        )
+
+        # Relationship between brine salinity and feed salinity
+        @self.Constraint(doc="Brine salinity")
```

Comment:
You should use the time index in most cases instead of just hard-coding 0 for the time index.